### PR TITLE
feat(body): added additional attribute '--body' to dsv secret create & dsv home c…

### PR DESCRIPTION
**Pull request by pacificcode**
added additional attribute '--body' to dsv secret create & dsv home.

Lint future work: [AB#561862](https://thycotic.visualstudio.com/4a89362e-1361-424f-a291-a8f57c2a8991/_workitems/edit/561862) [AB#561861](https://thycotic.visualstudio.com/4a89362e-1361-424f-a291-a8f57c2a8991/_workitems/edit/561861)

### CODE REVIEW

**Review by sheldonhull — COMMENTED** [review]
@pacificcode try running `trunk check` locally and should get 3 items flagged. Maybe existing code or not. If you need to add an exception you can document why that's a legit exception to bypass the check using the `//nolint:<linter> // documented reason why no lint shouldn't apply`.  Try to resolve if straight-forward, low effort, otherwise document why.

File: commands/secret.go [hunk 1]
```diff
@@ -510,20 +513,34 @@ func handleSecretUpsertCmd(vcli vaultcli.CLI, secretType string, action string,
...
-	if data != "" {
+	bodyMap := make(map[string]interface{})
+	/*Only process --body attribute if all 3 of the other attributes are empty*/
+	if body != "" && data == "" && desc == "" && len(attributes) == 0 {
+		parseErr := json.Unmarshal([]byte(body), &bodyMap)
+		if parseErr != nil {
+			vcli.Out().FailF("Failed to parse passed in secret body: %v", parseErr)
+			return 1
+		}
+		dataMap, _ = bodyMap["data"].(map[string]interface{})
```
Comment (andrii-zakurenyi) [inline commands/secret.go]: please also use here `any` instead of `interface{}`

**Review by andrii-zakurenyi — APPROVED** [review]
LGTM :+1:

Then only minor thing is that I think it is a good idea to use `any` instead of `interface{}`. If you update that then I'll re-approve.

**Review by sheldonhull — COMMENTED** [review]
For a failed checksum with Aqua in actions try Aqua update-checksum -prune and see if that file failed to get updated. It's a verification step so probably l missed being updated. 

Don't forget to run changie new for chabgekog entry as well.

**Comment by sheldonhull** [comment]
I see the test action failed. That's the first time I've ever seen that message and has to do with the signature for Aqua tool (a safety net for verifying the binary). I'll look at this and post an update today after checking into it.

File: .changes/unreleased/🎉 New Product Feature-20240321-125958.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+kind: "\U0001F389 New Product Feature"
+body:
+  "Added --body option to dsv secret create i.e. \n`dsv secret create --body '{\"data\":{\"key\":\"value\"},\"atrributes\":{\"attribute1\":\"value1\"},\"descr\":\"add
```
Comment (sheldonhull) [inline .changes/unreleased/🎉 New Product Feature-20240321-125958.yaml]:
```suggestion
  "Added `--body` option to dsv secret create i.e. \n`dsv secret create --body '{\"data\":{\"key\":\"value\"},\"atrributes\":{\"attribute1\":\"value1\"},\"descr\":\"add
```

File: commands/secret_test.go [hunk 3]
```diff
@@ -634,6 +650,22 @@ func TestHandleSecretUpsertCmd(t *testing.T) {
 			method:      "update",
 			expectedErr: nil,
 		},
+		{
+			name:        "Incorrect attributes in --body POST",
+			fDesc:       "new description",
+			args:        []string{"mySecret", "--body", "{\"datas\":{\"Key\":\"Values\"},\"attribute\":{\"Key2\":\"Value2\"},\"descs\":\"Secret description\"}"},
```
Comment (sheldonhull) [inline commands/secret_test.go:656]: is this mean to be `data` or `datas`

File: format/writers.go [hunk 4]
```diff
@@ -31,8 +31,9 @@ func NewFileWriter(filePath string) io.Writer {
 	return &fileWriter{filePath}
 }
 
+//nolint:gosec,gomnd // need to test this in a future PR
```
Comment (sheldonhull) [inline format/writers.go]: can you please create a bug/pbi and then just put AB#12345 to link to it. We try to make sure TODO's have future backlog captured so it doesn't disappear longterm.

File: utils/env.go [hunk 5]
```diff
@@ -12,8 +12,11 @@ type EnvProvider interface {
 }
 type envProvider struct{}
 
-var GetEnvProviderFunc func() EnvProvider
-var e EnvProvider
+//nolint:gochecknoglobals // need to test this in a future PR
```
Comment (sheldonhull) [inline utils/env.go]: same here, link to a future work-item for tracking for improving test coverage on this and link with AB#12345 format.

File: commands/secret_test.go (hunk 3 above)
Comment (pacificcode) [reply commands/secret_test.go:656]: yes, that is an error handling test

File: utils/env.go (hunk 5 above)
Comment (pacificcode) [reply utils/env.go]: https://thycotic.visualstudio.com/Delinea.Work/_workitems/edit/561862

File: format/writers.go (hunk 4 above)
Comment (pacificcode) [reply format/writers.go]: https://thycotic.visualstudio.com/Delinea.Work/_workitems/edit/561861

Comment (sheldonhull) [inline format/writers.go]:
```suggestion
//nolint:gosec,gomnd // TODO: AB#561861 need to test this in a future PR
```

@pacificcode this is what I meant. If todo make sure to note the todo. Azure board is automatically integrated with some of GitHub using AB# syntax and can even close the work items when you have in your PR "fixes AB#num"

Add the other comment please and note why bypassing gosec and we're good here. 

Thank you!

File: utils/env.go (hunk 5 above)
Comment (sheldonhull) [inline utils/env.go]: Add that other linked one here as TODO with AB# so it's clear and the quick why gosec is bypassed and all good.

Comment (sheldonhull) [reply utils/env.go]:
```suggestion
//nolint:gochecknoglobals // TODO: AB#561862 need to test this in a future PR
```

Comment (sheldonhull) [reply utils/env.go]: Commit these suggestions from ui and I'll approve. If it fails the SIEM test ask @andrii-zakurenyi who has more context and if he's cool with merge we'll do it. Cheers

**Review by sheldonhull — CHANGES_REQUESTED** [review]
pending batch commit changes and then this can be merged.

**Review by sheldonhull — APPROVED** [review]

**Pull request merged**